sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -165,28 +183,33 @@ def _create_handler(self, auth):
             encoding=self._config.encoding,
             receive_settle_mode=self._mode.value,
             send_settle_mode=SenderSettleMode.Settled if self._mode == ReceiveSettleMode.ReceiveAndDelete else None,
-            timeout=self._idle_timeout * 1000 if self._idle_timeout else 0,
-            prefetch=self._prefetch
+            timeout=self._max_wait_time * 1000 if self._max_wait_time else 0,
+            prefetch=self._prefetch,
+            shutdown_after_timeout=False
         )
 
     def _open(self):
         if self._running:
             return
-        if self._handler:
+        if self._handler and not self._handler._shutdown:
```

Comment:
Should we be closing if `self._handler._shutdown` is `True`? Or is this just an unfortunate backwards conditional? :P

Reply:
This was a response to the additional states the handler can be in, now that it does not shut down on timeout.  I interpreted the original logic as shutting down the handler if it was still active, and thus intentionally only did so now if it was not already shut down.  (Although now that you've got me thinking about it, truthfully I'd hope both close and open would be idempotent and we might not want to be proactively closing; or if we do it should be at the caller scope...)